# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/ffvpx/libavcodec/x86/flacdsp.asm
# Commit: 186e2a334f2d
# Full Hash: 186e2a334f2deb37b29f44b28bc8c83babeeb9a4
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-02-09 09:37:17
# Regressor Bug: 1877221
# File Overlap Count: 1
# Description:
#   Bug 1877221 - Pull in new ffmpeg code, reapply patch. r=media-playback-reviewers,chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D199909
# ==============================================================================

diff -r c5920de78d16 -r 186e2a334f2d media/ffvpx/libavcodec/x86/flacdsp.asm
--- a/media/ffvpx/libavcodec/x86/flacdsp.asm	Thu Feb 08 10:39:53 2024 +0000
+++ b/media/ffvpx/libavcodec/x86/flacdsp.asm	Thu Feb 08 10:39:54 2024 +0000
@@ -43,6 +43,7 @@
 cglobal flac_lpc_32, 5,6,5, decoded, coeffs, pred_order, qlevel, len, j
     sub    lend, pred_orderd
     jle .ret
+    movsxdifnidn pred_orderq, pred_orderd
     lea    decodedq, [decodedq+pred_orderq*4-8]
     lea    coeffsq, [coeffsq+pred_orderq*4]
     neg    pred_orderq